### PR TITLE
[[ Bug 19651 ]] Fix mirroring issues with video displayed using non-ARGB32 bitmap formats

### PR DESCRIPTION
**Pull request by livecodeian**

### CODE REVIEW

File: engine/src/w32-ds-player.cpp [hunk 1]
```diff
@@ -676,6 +735,71 @@ bool MCWin32DSPlayer::SetVisible(bool isVisible)
...
+		if (t_success)
+			t_success = SUCCEEDED(t_pin->ConnectedTo(&t_other_pin));
+
+		CComPtr<IEnumMediaTypes> t_mediatype_enum;
+		if (t_success)
+			t_success = SUCCEEDED(t_other_pin->EnumMediaTypes(&t_mediatype_enum));
+
+		AM_MEDIA_TYPE *t_mt = nil;
+		ULONG t_mt_fetched = 0;
+		while (t_success && (t_new_mediatype.majortype == MEDIATYPE_NULL) && (S_OK == t_mediatype_enum->Next(1, &t_mt, &t_mt_fetched)))
```
Comment (montegoulding) [inline engine/src/w32-ds-player.cpp]: @livecodeian probably not worth changing to clarify it as it's clear in the docs but you had me wondering for a bit if `t_mt_fetched` needed checking but it seems this param is ignored/can be null if `cMediaTypes` is 1.

File: engine/src/w32-ds-player.cpp [hunk 2]
```diff
@@ -762,8 +886,13 @@ bool MCWin32DSPlayer::OpenFile(MCStringRef p_filename)
 		}
 	}
 
-	if (t_success && t_vmr9 != NULL && !MCWin32DSFilterIsConnected(t_vmr9, PINDIR_INPUT))
-		t_success = SUCCEEDED(t_graph->RemoveFilter(t_vmr9));
+	if (t_success && t_vmr9 != NULL)
+	{
+		if (MCWin32DSFilterIsConnected(t_vmr9, PINDIR_INPUT))
+			/* UNCHECKED */ MCWin32DSReconnectVideoRendererConnection(t_graph, t_vmr9);
```
Comment (montegoulding) [inline engine/src/w32-ds-player.cpp:891]: @livecodeian any reason not to assign the result to `t_success` here?

Comment (montegoulding) [reply engine/src/w32-ds-player.cpp:891]: @livecodeian just checking if you saw the question here?

Comment (livecodeian) [reply engine/src/w32-ds-player.cpp:891]: Ah I think I assumed that MCWin32DSReconnectVideoRendererConnection could fail while still leaving the graph in a valid state - i.e. still able to play the video without the mirroring working

**Comment by montegoulding** [comment]
@livecode-vulcan review ok db2c96a

**Comment by livecode-vulcan** [comment]
💙 review by @montegoulding ok db2c96ac5722f590a1f821e6bffec3be64070509

**Comment by livecode-vulcan** [comment]
😎 test success db2c96ac5722f590a1f821e6bffec3be64070509

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/1483)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/1499)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/1473)
* try-community-universal-ios-iphoneos11.4: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos11.4/builds/0)
* try-community-universal-ios-iphonesimulator11.4: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator11.4/builds/0)
* try-community-universal-mac-macosx10.6: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/1477)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/1503)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/1483)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/1497)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/1482)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/1492)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/1532)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/1472)

**Pull request merged**